universal_login/deploy.ts

```diff
@@ -64,164 +79,45 @@ function loadEmail(emailName: Email): Record<string, unknown> {
+  instance: AxiosInstance,
+  prompt: Prompt
+): Promise<void> {
   const language = 'en';
-
   const promptData = loadPrompt(prompt);
-  const textPromptEndpoint = `https://${apiHost}/api/v2/prompts/${prompt}/custom-text/${language}`;
-
-  const promptUpdateRequest = {
-    method: 'PUT',
-    url: textPromptEndpoint,
-    headers: {
-      authorization: `Bearer ${token.access_token}`,
-      'content-type': 'application/json',
-    },
-    data: promptData,
-  } as AxiosRequestConfig;
-
-  await axios.request(promptUpdateRequest);
-}
-
-async function updateLoginPageTemplate(
-  env: Env,
-  template: Template,
-  token: BearerToken
-) {
-  const apiHost = getApiHost(env);
+  const textPromptEndpoint = `/api/v2/prompts/${prompt}/custom-text/${language}`;
 
-  const templateData = loadHTMLTemplate(template);
-  const templateEndpoint = `https://${apiHost}/api/v2/branding/templates/${template}`;
-
-  const templateUpdateRequest = {
-    method: 'PUT',
-    url: templateEndpoint,
-    headers: {
-      authorization: `Bearer ${token.access_token}`,
-      'content-type': 'text/html',
-    },
-    data: templateData,
-  } as AxiosRequestConfig;
-
-  await axios.request(templateUpdateRequest);
+  return instance.put(textPromptEndpoint, promptData);
 }
 
-async function updateEmail(env: Env, email: Email, token: BearerToken) {
-  const apiHost = getApiHost(env);
+function updateLoginPageTemplate(
+  instance: AxiosInstance,
+  template: Template
+): Promise<void> {
+  const templateData = loadHTMLTemplate(template);
+  const templateEndpoint = `/api/v2/branding/templates/${template}`;
 
-  const emailData = loadEmail(email);
-  const emailEndpoint = `https://${apiHost}/api/v2/email-templates/${email}`;
-
-  const templateUpdateRequest = {
-    method: 'PUT',
-    url: emailEndpoint,
-    headers: {
-      authorization: `Bearer ${token.access_token}`,
-      'content-type': 'application/json',
-    },
-    data: emailData,
-  } as AxiosRequestConfig;
-
-  await axios.request(templateUpdateRequest);
+  return instance.put(templateEndpoint, templateData, {
+    headers: { 'content-type': 'text/html' },
+  });
 }
 
-(async () => {
-  try {
-    const env = process.argv[2] as Env;
-    console.log(`Updating environment: ${env}`);
-
-    const token = await getApiToken(env);
-    // The non production rate limit of the API end point is 2 requests a second, this ensures we never hit the limit
-    const delay = () => new Promise((res) => setTimeout(res, 1000));
+async function updateEmail(
+  instance: AxiosInstance,
+  email: Email,
+  config: Record<string, string>
+): Promise<void> {
+  const emailData = loadEmail(email);
+  const emailEndpoint = `/api/v2/email-templates/${email}`;
 
-    for (const prompt of prompts) {
-      await delay();
-      console.log(`Updating ${prompt} prompt`);
-      await updateTextPrompt(env, prompt, token);
+  for (const [key, value] of Object.entries(emailData)) {
+    if (typeof value === 'string') {
```

Comment:
Is it worth logging a warning if we get a non-string value, or could that never happen?

Reply:
`Record<string, string>` should prevent that, TS doesn't seem to be able to reconcile that with `Object.entries`